### ISSUE TITLE
Unable to install under Archlinux

### ISSUE THREAD

**Issue by mantielero** [ThomasTJdev/nimlibxlsxwriter#1]
I am trying to install this under Linux but I get:
```
$ nimble install
   Warning: Package 'nimlibxlsxwriter' has an incorrect structure. It should contain a single directory hierarchy for source files, named 'nimlibxlsxwriter', but file 'doc_custom_properties.nim' is in a directory named 'tests' instead. This will be an error in the future.
      Hint: If 'tests' contains source files for building 'nimlibxlsxwriter', rename it to 'nimlibxlsxwriter'. Otherwise, prevent its installation by adding `skipDirs = @["tests"]` to the .nimble file.
Setting up Git repo: https://github.com/jmcnamara/libxlsxwriter
Pulling repository
Command failed: 1
git pull --depth=1 origin master
fatal: no se puedo encontrar ref remota master

stack trace: (most recent call last)
/tmp/nimblecache-1023837959/nimscriptapi_3203592219.nim(199, 29)
/tmp/nimlibxlsxwriter/nimlibxlsxwriter.nimble(24, 14) installBefore
/tmp/nimlibxlsxwriter/nimlibxlsxwriter.nimble(21, 10) setupTask
/home/jose/.choosenim/toolchains/nim-#devel/lib/system/nimscript.nim(273, 7) exec
/home/jose/.choosenim/toolchains/nim-#devel/lib/system/nimscript.nim(273, 7) Error: unhandled exception: FAILED: nimgen nimlibxlsxwriter.cfg [OSError]
       Tip: 3 messages have been suppressed, use --verbose to show them.
     Error: Exception raised during nimble script execution
```

I report the issue in the wrong repo: https://github.com/jmcnamara/libxlsxwriter/issues/329. Nonetheless, @jmcnamara was very helpful and suggested that the the issue is that rather than `master` is using `main`. I wonder if that can be handled with `nimgen`. Maybe `gitcheckout`?

**Comment by ThomasTJdev** [comment]
Hi @mantielero 

Great catch - thank you bringing this to my attention. You have identified the problem correct as the main branch has changed name.

In `nimgen` we need to use the `pull` procedure, and inside that procedure the branch name `master` is hardcoded. If you want to use the library before a fix to `nimgen` is published, then you can clone `nimgen` and change the name of branch. I have created an issue at `nimgen` here: https://github.com/genotrance/nimgen/issues/49. Otherwise a PR to `nimgen` would be nice :)

**Comment by ThomasTJdev** [comment]
Hi @mantielero 

I have created a PR for solving the branch name problem with `nimgen`. If you need to this library now, then:
1) Pull and `nimble install` the PR https://github.com/genotrance/nimgen/pull/50 
2) Update the `nimlibxlsxwriter` cfg-file with `gitbranch`:
```cfg
[n.prepare]
gitbranch = "main"
gitremote = "https://github.com/jmcnamara/libxlsxwriter"
gitsparse = """
include/*
include/xlsxwriter/*
"""
```
3) Do a nimble install within the `nimlibxlsxwriter` folder

**Comment by mantielero** [comment]
The installation works with your approach, thanks (I was looking how to do a PR).

**Comment by ThomasTJdev** [comment]
Closed by PR #5
